tests/test_cli/test_run.py

```diff
@@ -86,6 +87,442 @@ def test_run(pytestconfig):
+    assert result.exit_code == 0
+
+    os.chdir(Path(t, agent_name))
+
+    result = runner.invoke(cli, [*CLI_LOG_OPTION, "add", "connection", "local"])
+    assert result.exit_code == 0
+
+    # Load the agent yaml file and manually insert the things we need
+    file = open("aea-config.yaml", mode='r')
+
+    # read all lines at once
+    whole_file = file.read()
+
+    # add in the ledger address
+    find_text = "ledger_apis: []"
+    replace_text = """ledger_apis:
+        - ledger_api:
+            addr: https://ropsten.infura.io/v3/f00f7b3ba0e848ddbdc8941c527447fe
+            ledger: ethereum
+            port: 3
+        - ledger_api:
+            addr: alpha.fetch-ai.com
+            ledger: fetchai
+            port: 80"""
+
+    whole_file = whole_file.replace(find_text, replace_text)
+
+    # close the file
+    file.close()
+
+    with open("aea-config.yaml", 'w') as f:
+        f.write(whole_file)
+
+    with open("fet_private_key.txt", 'w') as f:
+        f.write("3801d3703a1fcef18f6bf393fba89245f36b175f4989d8d6e026300dad21e05d")
+
+    with open("eth_private_key.txt", 'w') as f:
+        f.write("0xa77f04041064441a5d944614ed82484a0852a29fdc1e8bbc88d80728f2d3f509")
+
+    process = subprocess.Popen([
+        sys.executable,
+        '-m',
+        'aea.cli',
+        "run",
+        "--connection",
+        "local"
+    ],
+        stdout=subprocess.PIPE,
+        env=os.environ.copy())
+
+    time.sleep(10.0)
+    process.send_signal(signal.SIGINT)
+    process.wait(timeout=20)
+
+    assert process.returncode == 0
+
+    os.chdir(cwd)
+
+    poll = process.poll()
+    if poll is None:
+        process.terminate()
+        process.wait(2)
+
+    try:
+        shutil.rmtree(t)
+    except (OSError, IOError):
+        pass
+
+
+def test_run_fet_ledger_apis(pytestconfig):
+    """Test that the command 'aea run' works as expected."""
+    if pytestconfig.getoption("ci"):
+        pytest.skip("Skipping the test since it doesn't work in CI.")
+
+    runner = CliRunner()
+    agent_name = "myagent"
+    cwd = os.getcwd()
+    t = tempfile.mkdtemp()
+    os.chdir(t)
+    result = runner.invoke(cli, [*CLI_LOG_OPTION, "create", agent_name])
+    assert result.exit_code == 0
+
+    os.chdir(Path(t, agent_name))
+
+    result = runner.invoke(cli, [*CLI_LOG_OPTION, "add", "connection", "local"])
+    assert result.exit_code == 0
+
+    # Load the agent yaml file and manually insert the things we need
+    file = open("aea-config.yaml", mode='r')
+
+    # read all lines at once
+    whole_file = file.read()
+
+    # add in the ledger address
+
+    find_text = "ledger_apis: []"
+    replace_text = """ledger_apis:
+    - ledger_api:
+        addr: alpha.fetch-ai.com
+        ledger: fetchai
+        port: 80"""
+
+    whole_file = whole_file.replace(find_text, replace_text)
+
+    # close the file
+    file.close()
+
+    with open("aea-config.yaml", 'w') as f:
+        f.write(whole_file)
+
+    with open("fet_private_key.txt", 'w') as f:
```

Comment:
Why are you creating the private keys? If it is to test the fact that they should already exist when `run` is called then please add a comment to make clear this is the case.

Reply:
done - and I could remove some of them